src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidInputException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int REQUIRED_AGE = 18;
+    private static final int MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidInputException("No data provided");
+        }
+        loginValidation(user);
+        passwordValidation(user);
+        ageValidation(user);
```

Comment:
you should use verbs instead of nouns in method names
```suggestion
        validateLogin(user);
        validatePassword(user);
        validateAge(user);
```